src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VariantsMongoReader.java

```diff
@@ -94,21 +104,33 @@ public void afterPropertiesSet() throws Exception {
     }
 
     @Override
-    protected void doOpen() throws Exception {
-        delegateReader.doOpen();
+    public void open(ExecutionContext executionContext) {
+        delegateReader.open(executionContext);
     }
 
     @Override
-    protected EnsemblVariant doRead() throws Exception {
-        DBObject dbObject = delegateReader.doRead();
-        if (dbObject != null) {
-            SimplifiedVariant variant = converter.read(SimplifiedVariant.class, dbObject);
-            return buildVariantWrapper(variant);
+    public List<EnsemblVariant> read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
+        DBObject firstElement = delegateReader.doRead();
+        if (firstElement != null) {
+            return fillBatch(firstElement);
         } else {
             return null;
         }
     }
 
+    private List<EnsemblVariant> fillBatch(DBObject dbObject) throws Exception {
+        List<EnsemblVariant> variants = new ArrayList<>();
+        while (dbObject != null) {
+            SimplifiedVariant variant = converter.read(SimplifiedVariant.class, dbObject);
+            variants.add(buildVariantWrapper(variant));
+            if (variants.size() == chunkSize) {
+                break;
+            }
+            dbObject = delegateReader.doRead();
```

Comment:
For me is confusing that the read() method is reading the first line and delegating the reading of the rest of the lines in the chunk to the fillBatch() one. The first time I read this class I thought it was returning a list with just one variant, and just in a second reading I noticed the real workflow. 

Can those two methods be refactored so is more clear that a chunk is being read and not just one element?